src/ai_hawk/job_manager.py

```diff
@@ -484,15 +490,27 @@ def extract_job_information_from_tile(self, job_tile):
 
     def is_blacklisted(self, job_title, company, link, job_location):
         logger.debug(f"Checking if job is blacklisted: {job_title} at {company} in {job_location}")
-        job_title_words = job_title.lower().split(' ')
-        title_blacklisted = any(word in job_title_words for word in map(str.lower, self.title_blacklist))
-        company_blacklisted = company.strip().lower() in (word.strip().lower() for word in self.company_blacklist)
-        location_blacklisted= job_location.strip().lower() in (word.strip().lower() for word in self.location_blacklist)
+        title_blacklisted = any(re.search(pattern, job_title, re.IGNORECASE) for pattern in self.title_blacklist_patterns)
+        company_blacklisted = any(re.search(pattern, company, re.IGNORECASE) for pattern in self.company_blacklist_patterns)
+        location_blacklisted = any(re.search(pattern, job_location, re.IGNORECASE) for pattern in self.location_blacklist_patterns)
         link_seen = link in self.seen_jobs
         is_blacklisted = title_blacklisted or company_blacklisted or location_blacklisted or link_seen
         logger.debug(f"Job blacklisted status: {is_blacklisted}")
 
-        return title_blacklisted or company_blacklisted or location_blacklisted or link_seen
+        return is_blacklisted
+
+    def generate_regex_patterns(self, blacklist):
```

Comment:
move this to regex_utils.py

Reply:
can you also add pyTest to this

Reply:
kindly rename method - demonstrating pattern goal/objective

Reply:
Should regex_utils.py be inside ai_hawk directory, or in source?
ill add pytest for it!

Reply:
i renamed the function to 'generate_regex_patterns_for_blacklisting'!